ObjectiveGit/GTIndex.h

```diff
@@ -135,6 +135,14 @@
 /// Returns YES if successful, NO otherwise.
 - (BOOL)addFile:(NSString *)file error:(NSError **)error;
 
+/// Add an entry (with the provided data and name) to the index.
+/// Will fail if the receiver's repository is nil.
+///
+/// data  - The content of the entry to add
```

Comment:
Looks like these can't be nil according to the implementation. Could we document that?